packages/web/src/components/gcds-radios/gcds-radios.tsx

```diff
@@ -0,0 +1,483 @@
+
+  @Watch('options')
+  validateOptions() {
+    let invalidObject = false;
+    // Assign optionsArr from passed options string or array
+    if (typeof this.options === 'string' && this.options.trim() !== '') {
+      try {
+        this.options = JSON.parse(this.options as string);
+      } catch (e) {
+        logError('gcds-radios', ['Invalid JSON string for options']);
+        this.options = null;
+      }
+    }
+
+    if (Array.isArray(this.options)) {
+      this.optionsArr = this.options;
+    } else {
+      this.optionsArr = null;
+    }
+
+    // Validate options has type RadioObject
+    if (this.optionsArr && this.optionsArr.length > 1) {
+      invalidObject = this.optionsArr.some(radio => !isRadioObject(radio));
+    } else {
+      invalidObject = true;
+    }
+
+    // Assign value if passed options has a checked radio
+    if (this.optionsArr && !this.value) {
+      this.optionsArr.forEach(radio => {
+        if (radio.checked === 'true' || radio.checked === true) {
+          this.value = radio.value;
+          this.internals.setFormValue(radio.value, 'checked');
+        }
+      });
+    }
+
+    // Log error if no or invalid optionsObject
+    this.errors = handleErrors(
+      this.errors,
+      'options',
+      this.optionsArr,
+      invalidObject,
+    );
+  }
+
+  /**
+   * The `name` attribute for the <gcds-radios>, used to group radio elements together
+   */
+  @Prop({ reflect: true, mutable: false }) name!: string;
+
+  @Watch('name')
+  validateName() {
+    this.errors = handleErrors(this.errors, 'name', this.name);
+  }
+
+  /**
+   * Label or legend for the group of radio elements
+   */
+  @Prop({ reflect: true, mutable: false }) legend!: string;
+  @Watch('legend')
+  validateLegend() {
+    this.errors = handleErrors(this.errors, 'legend', this.legend);
+  }
+
+  /**
+   * Specifies if a form field is required or not.
+   */
+  @Prop({ reflect: true, mutable: false }) required: boolean;
+
+  /**
+   * Hint displayed below the label and above the radio elements
+   */
+  @Prop({ reflect: true, mutable: false }) hint: string;
+
+  /**
+   * Set this to display an error message for invalid <gcds-radios>
+   */
+  @Prop({ reflect: false, mutable: true }) errorMessage: string;
+  @Watch('errorMessage')
+  validateErrorMessage() {
+    if (this.disabled) {
+      this.errorMessage = '';
+    } else {
+      this.hasError = this.errorMessage ? !this.hasError : false;
+    }
+  }
+
+  /**
+   * Specifies if an input element is disabled or not.
+   */
+  @Prop({ reflect: true, mutable: true }) disabled: boolean;
+
+  /**
+   * Default value for the element
+   */
+  @Prop({ reflect: true, mutable: true }) value: string;
+  @Watch('value')
+  validateValue() {
+    if (this.optionsArr && this.value !== null) {
+      let isValidValue = false;
+      this.optionsArr.map(radio => {
+        if (radio.value == this.value) {
+          isValidValue = true;
+          radio.checked = true;
```

Comment:
Do we need to mutate `radio.checked` here, is it possible to remove this? We're already updating if it's checked or not in this line: https://github.com/cds-snc/gcds-components/pull/845/files#diff-6cd0074b871d8bc998ce3ef3c019f185e4afcde7fab603bc53e79e0d257237f3R431

Is that not enough?